database-connector/src/main/java/life/qbic/projectmanagement/persistence/repository/OpenbisConnector.java

```diff
@@ -131,32 +180,167 @@ public List<Analyte> retrieveAnalytes() {
+    IOperation operation = new CreateSamplesOperation(samplesToRegister);
+    handleOperations(operation);
+  }
+
+  /**
+   * Deletes a sample with the provided code from persistence.
+   *
+   * @param sampleCode the {@link SampleCode} of the sample to delete
+   * @since 1.0.0
+   */
+  @Override
+  public void delete(SampleCode sampleCode) {
+    deleteOpenbisSample(DEFAULT_SPACE_CODE, sampleCode.code());
+  }
+
+  /**
+   * Searches for samples that contain the provided sample code
+   *
+   * @param sampleCode the {@link SampleCode} to search for in the data repository
+   * @return true, if a sample with that code already exists in the system, false if not
+   * @since 1.0.0
+   */
+  @Override
+  public boolean sampleExists(SampleCode sampleCode) {
+    return !searchSamplesByCode(sampleCode.code()).isEmpty();
+  }
+
   record VocabularyTerm(String code, String label, String description) {
 
   }
 
-  private void createOpenbisProject(String spaceCodeString, ProjectCode projectCode) {
+  private void createOpenbisProject(String spaceCodeString, String projectCode) {
     ProjectCreation project = new ProjectCreation();
-    project.setCode(projectCode.value());
+    project.setCode(projectCode);
     project.setSpaceId(new SpacePermId(spaceCodeString));
 
     IOperation operation = new CreateProjectsOperation(project);
     handleOperations(operation);
   }
 
-  private void deleteOpenbisProject(ProjectCode projectCode) {
+  private void createOpenbisExperiment(String spaceCode, String projectCode,
+      String experimentCode) {
+    ExperimentCreation experiment = new ExperimentCreation();
+    experiment.setTypeId(new EntityTypePermId("Q_SAMPLE_PREPARATION"));
+    experiment.setProjectId(new ProjectIdentifier(spaceCode, projectCode));
+    experiment.setCode(experimentCode);
+
+    IOperation operation = new CreateExperimentsOperation(experiment);
+    handleOperations(operation);
+  }
+
+  private void deleteOpenbisProject(String spaceCode, String projectCode) {
     ProjectDeletionOptions deletionOptions = new ProjectDeletionOptions();
     deletionOptions.setReason("unknown reason in data-manager");
     //OpenBis expects the projectspace and code during deletion
-    ProjectIdentifier projectIdentifier = new ProjectIdentifier(DEFAULT_SPACE_CODE,
-        projectCode.value());
+    ProjectIdentifier projectIdentifier = new ProjectIdentifier(spaceCode,
+        projectCode);
     List<ProjectIdentifier> openBisProjectsIds = new ArrayList<>();
     openBisProjectsIds.add(projectIdentifier);
     DeleteProjectsOperation operation = new DeleteProjectsOperation(openBisProjectsIds,
         deletionOptions);
     handleOperations(operation);
   }
 
+  private void deleteOpenbisSample(String spaceCode, String sampleCode) {
+    SampleDeletionOptions deletionOptions = new SampleDeletionOptions();
+    deletionOptions.setReason("unknown reason in data-manager");
```

Comment:
I am not sure if I get the reason :D

Reply:
ah I see. However it not necessary means failing, since we can also delete samples intentionally (faulty batch registration by the user, that they want to revert).

I suggest to write "Commanded by data manager app" and introduce a constant field, since this might get extended.